2-WebApp-graph-user/2-3-Multi-Tenant/Controllers/OnboardingController.cs

```diff
@@ -50,6 +50,7 @@ public OnboardingController(SampleDbContext dbContext, IOptions<MicrosoftIdentit
             this.dbContext = dbContext;
             this.microsoftIdentityOptions = microsoftIdentityOptions.Value;
             this.configuration = configuration;
+            configuration.Bind("AzureAd", this.microsoftIdentityOptions);
```

Comment:
This shouldn't be necessary. This should be done in the Startup.cs:
https://github.com/Azure-Samples/active-directory-aspnetcore-webapp-openidconnect-v2/blob/8c3aef9ef5963501efba76151cd1891ebc7e8b90/2-WebApp-graph-user/2-3-Multi-Tenant/Startup.cs#L62

Which problem is your PR solving? do you have repro steps?
(what is expected, what is observed?)

Reply:
The problem there is that 
`this.microsoftIdentityOptions = microsoftIdentityOptions.Value;`
doesn't initialize all the data - the required fields are empty. 
To reproduce: run the sample as is without the fix, try onboard a tenant and you will get exception on empty field while trying to construct Consent URL.

Reply:
Thanks @aremo-ms 
I think that the fix would be to inject an `IOptionsMonitor<MicrosoftIdentityOptions>`, and get it for the current authentication scheme (OpenIdConnectDefault.AuthenticationScheme)

cc: @jennyf19

Reply:
@aremo-ms:
The right fix is to:
1. Inject an `IOptionsMonitor<MicrosoftIdentityOptions>` instead of `IOptionsMonitor<MicrosoftIdentityOptions>` in the constructor of `OnboardingController`
2. initialiaz the `microsoftIdentityOptions` as  the value of the IOptionsMonitor for  the specific authentication scheme `OpenIdConnectDefaults.AuthenticationScheme)`

```CSharp
        public OnboardingController(SampleDbContext dbContext, 
                                                       IOptions<MicrosoftIdentityOptions> microsoftIdentityOptions, 
                                                       IConfiguration configuration)
        {
            this.dbContext = dbContext;
            this.microsoftIdentityOptions = microsoftIdentityOptions.Value;
            this.configuration = configuration;
        }
```

Therefore we need to have

```CSharp
        public OnboardingController(SampleDbContext dbContext, 
                                                       IOptionsMonitor<MicrosoftIdentityOptions> microsoftIdentityOptions,
                                                       IConfiguration configuration)
        {
            this.dbContext = dbContext;
            this.microsoftIdentityOptions = microsoftIdentityOptions.Get(OpenIdConnectDefaults.AuthenticationScheme);
            this.configuration = configuration;
        }
```

This regression was introduced in MicrosoftIdentityWeb 1.11.0, when Microsoft.Identity.Web started supporting multiple authentication schemes. We've taken a lot of precautions to ensure the backward compatibility, but we didn't think of this scenario.

cc: @jennyf19 
I've updated the wiki https://github.com/AzureAD/microsoft-identity-web/wiki/Multiple-Authentication-Schemes#backward-compatibility